src/TMTProductizer/Services/TMT/TMTAPIAuthorizationService.cs

```diff
@@ -7,39 +7,46 @@
 
 namespace TMTProductizer.Services.TMT;
 
-public class TMTAuthorizationService : ITMTAuthorizationService
+public class TMTAPIAuthorizationService : ITMTAPIAuthorizationService
 {
     private readonly HttpClient _client;
-    private readonly ISecretsManager _secretsManager;
-    private readonly ILogger<TMTAuthorizationService> _logger;
-    private TMTAuthorizationDetails? _TMTAuthorizationDetails = null;
+    private readonly IDynamoDBCache _dynamoDBCache;
+    private readonly ITMTSecretsManager _secretsManager;
+    private readonly ILogger<TMTAPIAuthorizationService> _logger;
+    private TMTAPIAuthorizationDetails? _TMTAPIAuthorizationDetails = null;
     private bool _skipAuthorizationCeck;
+    private const string _cacheKey = "TMTAPIAuthorizationDetails";
```

Comment:
Kysymys kun en tunne dynamoDB niin hyvin:

Onko dynamo DB:n avaimet globaaleja kuten rediksessä? Tai ainakin näin muistaisin että oli kun viimeksi käytin.

Pohdin vain että voiko samalla avaimella joku tallentaa eri dataa joko samalla schemalla jolloin voisi korvaantua dataa jota ei haluta, tai sitten eri schemalla ja sitten kun haet datan ja se muutetaan takaisin objektiksi niin saat jonkin sortin deserialize exceptionin.

Ei välttämättä ole valiidi dynamoDB:ssä ja sitten jos tämän omistaa vain tämä palvelu niin vain palvelu itsessään voi tehdä jotain väärää. Edellisessä projektissa jossa olin niin rediksen kanssa tehtiin oma cache key logiikka joka yksilöi avaimen niin että ei voinut tulla yhteen törmäyksiä, eli tyylin cache avain + joku muut uniikki tunniste liittyen serialisoitavaan obektiin + invalidointi sääntöjä kuten aikarajat jotka taitaa olla varmaankin tässä PR:ssä.

Reply:
Dynamodb:n avaimet ovat taulun kohdalla "globaaleja" eli jos joku toinen palvelu käyttäisi samaa taulua niin törmäily olisi ihan mahdollista, mutta ehkäistävissä esim. pakotetuilla tietorakenteilla ja tietty generoidulla avaimella.  Tässä tapauksessa ajatus on että cache on omisteinen vain tälle palvelulle, mutta totta on että jos sitä jatkokehittelyssä käyttelee niin on mahdollisuus osua ojaan varsinkin kun tässä cachessa ajatuksena on ettei scheemaa tsekkailla kanta-tasolla vaan kantaan menee pelkästään tekstiä. 

Laitoin pieneksi ehkäisyksi sellaisen että avain generoidaan nimestä ja tiedon tyypistä mallia: https://github.com/Virtual-Finland-Development/tmt-productizer/blob/VFD-115-valimuistipalvelu-lambda-funkkareiden-kayttoon/src/TMTProductizer.UnitTests/UtilityTests.cs#L16